pkg/prometheus_ingester/headerRoundTripper.go

```diff
@@ -0,0 +1,30 @@
+package prometheus_ingester
+
+import "net/http"
+
+type httpHeadersRoundTripper struct {
+	headers map[string]string
+	rt      http.RoundTripper
+}
+
+// RoundTrip implements the http.RoundTripper interface.
+func (h httpHeadersRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	// The specification of http.RoundTripper says that it shouldn't mutate
+	// the request so make a copy of req.Header since this is all that is
+	// modified. https://pkg.go.dev/net/http#RoundTripper
+	r2 := new(http.Request)
+	*r2 = *r
+	r2.Header = make(http.Header)
+	// copy existing headers
+	for k, s := range r.Header {
```

Comment:
Why "k, s"? "k, v" seem to be the usual names, which you do use further down.
```suggestion
	for k, v := range r.Header {
```

Reply:
fixed 7e203566904347b1ed7c94a78e30ea85dddfafa9